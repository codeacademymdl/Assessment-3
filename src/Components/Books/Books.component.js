import React, { Component } from 'react';
import './Books.css';
import BlackHeartImage from '../../Icons/heart-black.svg'
import RedHeartImage from '../../Icons/heart-red.svg';

class Books extends Component {


    state = {
        isLike: true,
    }
    likeDislike = () => {

        this.setState({
            isLike: !this.state.isLike
        })

    }
    getImage = () => this.state.isLike ? BlackHeartImage : RedHeartImage;
    cardColor= ()=>this.props.colorIndex%2===0 ?"#FC9152":"white"
        
     
    render() {

        return (
            <div className="book" style={{backgroundColor:this.cardColor()}} >
                <img src={this.props.imageSrc} alt="book" className='image' />
                <div className="container">

                    <p className='title'>{this.props.Name}</p>

                    <span>{this.props.rating}</span>

                    <img alt="" className="heart" src={this.getImage()} onClick={this.likeDislike} />
                </div>
            </div>
        );
    }
}

export default Books;
