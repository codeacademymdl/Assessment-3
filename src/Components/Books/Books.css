.book {

    text-align: center;
    /* background-color:#FC9152; */
    /* margin-left: 3%;
    margin-right: 3%; */
    
    width: 500px;
    margin:auto;
    margin-top: 2%;
    height:750px;
  }

  .heart {
    width: 10%; 
    float: right;
  }
  
  .image {
    width: 100%;

  }

  .title {
    font-size: 1em;
    text-align: center;
  }
  
  .container {
    text-align: left;
    font-family: Arial, Helvetica, sans-serif;
    font-size:1em;
    padding:10px;
  }