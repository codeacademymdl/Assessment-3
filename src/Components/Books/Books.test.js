import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Books from './Books.component'

const book = {
    "Name": "Harry Potter and the Sorcerers Stone (Harry Potter, #1)",
    "rating": "4.45",
    "imageSrc": "{`./cover_image.jpeg`}"
}
describe('Books', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Books />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});



describe('Book likeDislike', () => {
    it('checks if state of islike is changed ', () => {
        const wrapper = shallow(<Books
            Name="Harry Potter and the Sorcerers Stone (Harry Potter, #1)"
            rating="4.45"
            imageSrc= "{`./cover_image.jpeg`}"

            />)
        expect(wrapper.instance().state.isLike).toEqual(true);
        wrapper.instance().likeDislike();
        expect(wrapper.instance().state.isLike).toEqual(!true);

    });

});


