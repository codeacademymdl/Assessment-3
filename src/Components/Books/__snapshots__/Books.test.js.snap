// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books renders correctly 1`] = `
<div
  className="book"
>
  <img
    alt="book"
    className="image"
  />
  <div
    className="container"
  >
    <p
      className="title"
    />
    <span />
    <img
      alt=""
      className="heart"
      onClick={[Function]}
      src="heart-black.svg"
    />
  </div>
</div>
`;
